packages/ui/src/components/Chip/index.tsx

```diff
@@ -0,0 +1,142 @@
+import { Stack } from '../Stack'
+import { ChipContext } from './ChipContext'
+import { ChipIcon } from './ChipIcon'
+
+const StyledContainer = styled(Stack)`
+  padding: ${({ theme }) => `${theme.space['0.5']} ${theme.space['2']} `};
+  display: flex;
+  border-radius: ${({ theme }) => theme.radii.xlarge};
+  width: fit-content;
+  background-color: ${({ theme }) => theme.colors.neutral.background};
+  cursor: pointer;
+  border: 1px solid ${({ theme }) => theme.colors.neutral.borderWeak};
+  text-align: center;
+  color: ${({ theme }) => theme.colors.neutral.text};
+  user-select: none;
+
+  &[data-disabled="false"]:hover {
+    background-color: ${({ theme }) => theme.colors.neutral.backgroundHover};
+    border-color: ${({ theme }) => theme.colors.neutral.borderStrongHover};
+    color: ${({ theme }) => theme.colors.neutral.textHover};
+
+  }
+
+  &[data-disabled="true"] {
+    background-color: ${({ theme }) => theme.colors.neutral.backgroundDisabled};
+    border-color: ${({ theme }) => theme.colors.neutral.borderWeakDisabled};
+    color: ${({ theme }) => theme.colors.neutral.textDisabled};
+    cursor: not-allowed;
+  }
+
+  &[data-active="true"]{
+    background-color: ${({ theme }) => theme.colors.primary.backgroundStrong};
+    border-color:  ${({ theme }) => theme.colors.primary.backgroundStrong};
+    color: ${({ theme }) => theme.colors.neutral.textStronger};
+
+    &[data-disabled="false"]:hover{
+      background-color: ${({ theme }) => theme.colors.primary.backgroundStrongHover};
+      border-color: ${({ theme }) => theme.colors.primary.backgroundStrongHover};
+      color: ${({ theme }) => theme.colors.neutral.textStrongerHover};
+
+
+    }
+
+    &[data-disabled="true"] {
+      background-color: ${({ theme }) => theme.colors.primary.backgroundStrongDisabled};
+      border: none;
+    }
+  }
+
+  &[data-size='medium']{
+    ${({ theme }) =>
+      `font-size: ${theme.typography.caption.fontSize};
+    font-family: ${theme.typography.caption.fontFamily};
+    font-weight: ${theme.typography.caption.weight};
+    letter-spacing: ${theme.typography.caption.letterSpacing};
+    line-height: ${theme.typography.caption.lineHeight};
+    text-transform: ${theme.typography.caption.textCase};
+    text-decoration: ${theme.typography.caption.textDecoration};
+    height: ${theme.space[3]};
+    padding: ${theme.space['0.5']} ${theme.space['1.5']};`}
+  }
+
+  &[data-size='large']{
+    ${({ theme }) =>
+      `font-size: ${theme.typography.bodySmall.fontSize};
+    font-family: ${theme.typography.bodySmall.fontFamily};
+    font-weight: ${theme.typography.bodySmall.weight};
+    letter-spacing: ${theme.typography.bodySmall.letterSpacing};
+    line-height: ${theme.typography.bodySmall.lineHeight};
+    text-transform: ${theme.typography.bodySmall.textCase};
+    text-decoration: ${theme.typography.bodySmall.textDecoration};
+    height: ${theme.space[4]};
+    padding: ${theme.space['0.5']} ${theme.space['2']};`}
+
+  }
```

Comment:
You don't need to redo all of this, instead wrap the children into a `Text` component and set the `as` prop to `text` if `typeof children === string` and to div if not. This will apply the same properties. Don't forget to set the variant depending on the size.